### PR TITLE
fixes for dropdownmenu and rule builder

### PR DESCRIPTION
**Pull request by MaxBittker**
addresses david's concerns, updates storybook, fixes keyboard highlight for grouped usecase

### CODE REVIEW

File: docs-ui/components/dropdownAutoComplete.stories.js [hunk 1]
```diff
@@ -23,17 +23,15 @@ const items = [
...
-      label: (
-        <div>
-          Countries{' '}
-          <a style={{float: 'right'}} href="#">
-            + Add
-          </a>
-        </div>
-      ),
-    },
+    value: 'countries',
```
Comment (MaxBittker) [inline docs-ui/components/dropdownAutoComplete.stories.js]: updates the shape in the stories

File: docs-ui/components/dropdownAutoComplete.stories.js [hunk 2]
```diff
@@ -75,23 +71,23 @@ storiesOf('DropdownAutoComplete', module)
   .add(
     'ungrouped',
     withInfo('The item label can be a component or a string')(() => (
-      <DropdownAutoComplete items={items}>
+      <DropdownAutoComplete items={items} alignMenu="left">
```
Comment (MaxBittker) [inline docs-ui/components/dropdownAutoComplete.stories.js]: stories worked better with this alignment so they weren't cut off

File: src/sentry/static/sentry/app/components/autoComplete.jsx [hunk 3]
```diff
@@ -185,7 +185,7 @@ class AutoComplete extends React.Component {
       onOpen(...args);
       return;
     }
-
+    this.resetHighlightState();
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/autoComplete.jsx]: highlight state is weird when it persists across open/close imo

File: src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx [hunk 4]
```diff
@@ -153,15 +153,15 @@ class DropdownAutoCompleteMenu extends React.Component {
                   <div>
                     {menuHeader && <StyledLabel>{menuHeader}</StyledLabel>}
                     {this.autoCompleteFilter(items, inputValue).map(
-                      (item, index) =>
+                      ({index, ...item}) =>
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx]: we want to use our special index that gets annotated in, not the one from map

File: src/sentry/static/sentry/app/views/settings/project/projectOwnership/ruleBuilder.jsx [hunk 5]
```diff
@@ -116,7 +116,7 @@ class RuleBuilder extends React.Component {
     return projectData.teams.map(team => ({
       value: buildTeamId(team.id),
       label: `#${team.slug}`,
-      searchKey: team.slug,
+      searchKey: `#${team.slug}`,
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/settings/project/projectOwnership/ruleBuilder.jsx]: make these searchable with #

File: src/sentry/static/sentry/app/components/autoComplete.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/autoComplete.jsx]: Agreed, thought this was weird

File: src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx (hunk 4 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx]: Oops I think I broke this, we should probably add a test case for this

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/autoComplete.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/autoComplete.jsx]: Can you add a test for this?

**Pull request merged**